mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -89,12 +93,30 @@ impl S3CrtClient {
 
         let mut client_bootstrap = ClientBootstrap::new(&allocator, &bootstrap_options).unwrap();
 
-        let creds_options = CredentialsProviderChainDefaultOptions {
-            bootstrap: &mut client_bootstrap,
-        };
+        let mut client_config = ClientConfig::new();
 
-        let mut creds_provider = CredentialsProvider::new_chain_default(&allocator, &creds_options).unwrap();
-        let signing_config = init_default_signing_config(region, &mut creds_provider);
+        if !config.no_sign_request {
+            let credentials_provider = match config.profile_name_override {
+                Some(profile_name_override) => {
+                    let credentials_profile_options = CredentialsProviderProfileOptions {
+                        bootstrap: &mut client_bootstrap,
+                        profile_name_override: Some(&profile_name_override),
+                    };
+                    CredentialsProvider::new_profile(&allocator, credentials_profile_options)
+                        .map_err(|_| CredentialsError::ProfileNotFound)
+                }
+                None => {
+                    let credentials_chain_default_options = CredentialsProviderChainDefaultOptions {
+                        bootstrap: &mut client_bootstrap,
+                    };
+                    CredentialsProvider::new_chain_default(&allocator, credentials_chain_default_options)
+                        .map_err(|_| CredentialsError::NoValidCredentials)
```

Comment:
Same thing here about the error:
1. `CredentialsError::ProviderFailure`
2. don't throw away the underlying error